File: StartAppFrontend/app/src/main/java/com/skyletto/startappfrontend/data/database/AppDatabase.java
package com.skyletto.startappfrontend.data.database;

import android.content.Context;

import com.skyletto.startappfrontend.domain.entities.Tag;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Tag.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase db;
    private static final String DB_NAME = "main.db";

    public static AppDatabase getInstance(Context context){
        if (db ==null)
            synchronized (AppDatabase.class) {
                if (db == null)
                    db = Room.databaseBuilder(context, AppDatabase.class, DB_NAME).build();
            }
        return db;
    }
}

File: StartAppFrontend/app/src/main/java/com/skyletto/startappfrontend/data/database/AppDatabase.kt
package com.skyletto.startappfrontend.data.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.skyletto.startappfrontend.common.models.UserTags
import com.skyletto.startappfrontend.data.database.dao.*
import com.skyletto.startappfrontend.domain.entities.*

@Database(entities = [Tag::class, Message::class, Chat::class, User::class, UserTags::class, ProjectRole::class], version = 4, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun tagDao(): TagDao
    abstract fun messageDao(): MessageDao
    abstract fun chatDao(): ChatDao
    abstract fun userDao(): UserDao
    abstract fun userTagsDao(): UserTagsDao
    abstract fun roleDao(): RoleDao

    companion object {
        private var db: AppDatabase? = null
        private const val DB_NAME = "main.db"
        @JvmStatic
        fun getInstance(context: Context): AppDatabase {
            if (db == null)
                synchronized(AppDatabase::class.java) {
                if (db == null)
                    db = Room.databaseBuilder(context, AppDatabase::class.java, DB_NAME).fallbackToDestructiveMigration().build()
            }
            return db!!
        }
    }
}